dia/compare-anonymous-appraisal.rkt

```diff
@@ -0,0 +1,50 @@
+#lang racket
+
+;; Run with `racket compare-anonymous-appraisal.rkt`
+;; or `racket -l- abe/compare-anonymous-appraisal`
+
+(require "tree-parser.rkt")
+
+(define (get-tree ip)
+  (when (regexp-match-peek #rx"#lang" ip)
+    (read-line ip 'any))
+  (read-anonymous-attribution-tree2 ip))
+
+(define (make-rows trees)
+  (let ([trees (map flatten trees)])
+    (reverse
+     (let loop ([result '()]
+                [trees trees])
+       (match trees
+         [(list '() ...) result]
+         [(list (cons (? string? x) rest) ...)
+          #:when (apply equal? x)
```

Comment:
This will error there are not exactly 2: unlike the specialized equality tests, `equal?` is strictly binary/dyadic.

Reply:
Ah yeah, I assumed it was variadic too. A simple fix could be to use `=` from `(require relation)`...

Reply:
I think `string=?` will do it.

Reply:
Ran into this:

```
$ racket compare-anonymous-appraisal.rkt ../../dia-old-abe/evaluation/appraisal/ben/capital.md ../../dia-old-abe/evaluation/appraisal/sid/capital.md ../../dia-old-abe/evaluation/appraisal/robin/capital.md
match: no matching clause for '(("Configuration template allowing the project to declare dependencies and other packaging metadata. " 10 "Hosting" 20 "The infrastructure that runs tests and other code quality checks during development. " 35 "The platform publicly hosting the source ...
  location...:
   compare-anonymous-appraisal.rkt:18:7
  context...:
   /Applications/Racket-Latest/collects/racket/match/gen-match.rkt:90:24: fail2
   /Users/siddhartha/work/abe/dia/dia/compare-anonymous-appraisal.rkt:13:0: make-rows
   /Applications/Racket-Latest/collects/racket/cmdline.rkt:191:51
   body of (submod "/Users/siddhartha/work/abe/dia/dia/compare-anonymous-appraisal.rkt" main)
```

@apromessi 's appraisal isn't compared here since it's in a different format that she recommended some time ago, of putting the numbers in the leading position so that they are aligned and easy to compare visually. We haven't discussed this yet but we should at some point.

Reply:
Yeah, haven't tested on Robin or Ariana's files yet. I expect I'll need to tweak them to make it work.

Reply:
Found it. Robin's file is missing a space:
```diff
diff --git i/evaluation/appraisal/robin/capital.md w/evaluation/appraisal/robin/capital.md
index 6a45647..f8a6d37 100644
--- i/evaluation/appraisal/robin/capital.md
+++ w/evaluation/appraisal/robin/capital.md
@@ -17,7 +17,7 @@
       * The programming language that the project is implemented in. [25%]
       * The language in which the project's documentation is written. [25%]
       * A unit-testing framework for defining and running tests to verify project functionality during development. [30%]
-      * Configuration template allowing the project to declare dependencies and other packaging metadata.[20%]
+      * Configuration template allowing the project to declare dependencies and other packaging metadata. [20%]
     * Hosting [15%]
       * The infrastructure that runs tests and other code quality checks during development. [35%]
       * The platform publicly hosting the source code repository. [30%]
```

The error there could certainly be better :)